test/test_helper.rb

```diff
@@ -25,7 +25,19 @@
 
 module Minitest
   class Test
+    extend T::Sig
+
     Minitest::Test.make_my_diffs_pretty!
+
+    sig { void }
+    def stub_typechecking
+      RubyLsp::DependencyDetector.instance.stubs(:typechecker?).returns(false)
+    end
+
+    sig { void }
+    def teardown
+      Singleton.__init__(RubyLsp::DependencyDetector)
+    end
```

Comment:
> Are there too many tests that depend on the state of the singleton being reset? Can we reset the singleton on each test that requires it instead?

We could do that, but I think has a similar problem that if we introduce a new test, then we need to remember to add that teardown. And since it's order dependant, CI may still pass.

Another approach would be to switch the tests to use `ActiveSupport::TestCase`. That provides a `teardown {}` method that takes a block, which automatically calls `super`. I know it's another dependency to add, but it would only be required for tests.

Reply:
IMO, let's not add this teardown. The test for the dependency detector itself can do this in its own teardown.

For all other tests that depend on a specific value being returned by something, we should use stubbing.